src/Controller/ExportController.php

```diff
@@ -29,24 +31,48 @@ class ExportController extends AppController
     const DEFAULT_EXPORT_LIMIT = 10000;
 
     /**
-     * Export data in csv format
+     * {@inheritDoc}
+     * {@codeCoverageIgnore}
+     */
+    public function initialize(): void
+    {
+        parent::initialize();
+
+        $this->loadComponent('Export');
+    }
+
+    /**
+     * Export data to format specified by user
      *
-     * @return \Cake\Http\Response
+     * @return \Cake\Http\Response|null
      */
-    public function export(): Response
+    public function export(): ?Response
     {
+        // allow only specific export formats: csv, ods, xlsx
+        $format = (string)$this->request->getData('format');
+        if (!in_array($format, ['csv', 'ods', 'xlsx'])) {
```

Comment:
use a `const ALLOWED_FORMATS` instead of `['csv', 'ods', 'xlsx']` array? 

format list could change, also using plugins for instance